tests/core_tests/test_context.py

```diff
@@ -434,8 +434,83 @@ def setUp(self):
                      "project": self.project,
                      "entity": self.shot,
                      "step": self.step}
+
+        self.publishedfile = dict(
+            id=2,
+            type="PublishedFile",
+            project=self.project,
+            entity=self.shot,
+            task=self.task,
+        )
+
+        self.version = dict(
+            id=3,
+            type="Version",
+            project=self.project,
+            entity=self.shot,
+        )
         
         self.add_to_sg_mock_db(self.task)
+        self.add_to_sg_mock_db(self.publishedfile)
+        self.add_to_sg_mock_db(self.version)
+
+    @patch("tank.util.login.get_current_user")
+    def test_from_linked_entity_types(self, get_current_user):
+        get_current_user.return_value = self.current_user
+
+        # PublishedFile and Version entities are special cased in the context
+        # factories. We need to make sure they create a context object that is
+        # built from what those entities are linked to, but with the original
+        # entity kept as the source_entity of the context.
+        result = context.from_entity(self.tk, self.publishedfile["type"], self.publishedfile["id"])
+        self.check_entity(self.project, result.project, check_name=False)
+        self.check_entity(self.shot, result.entity, check_name=False)
+        self.check_entity(self.task, result.task, check_name=False)
+        self.check_entity(
+            result.source_entity,
+            dict(
+                type=self.publishedfile["type"],
+                id=self.publishedfile["id"],
+            ),
+            check_name=False,
+        )
+
+        result = context.from_entity(self.tk, self.version["type"], self.version["id"])
+        self.check_entity(self.project, result.project, check_name=False)
+        self.check_entity(self.shot, result.entity, check_name=False)
+        self.check_entity(
+            result.source_entity,
+            dict(
+                type=self.version["type"],
+                id=self.version["id"],
+            ),
+            check_name=False,
```

Comment:
Why would you not check for the name? Is it because you're not copying it? If not, why aren't we? Are we not copying it for other entities in the context?